### PR TITLE
[ticket/15553] Add method to get direct link

### PR DESCRIPTION
**Pull request by rubencm**
PHPBB3-15553

Checklist:

- [x] Correct branch: master for new features; 3.2.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/dev/master/development/coding_guidelines.html) and [3.2.x](https://area51.phpbb.com/docs/dev/3.2.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.2.x/development/git.html)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-15553

### CODE REVIEW

File: phpBB/phpbb/storage/adapter/local.php [hunk 1]
```diff
@@ -331,4 +337,12 @@ public function file_image_height($path)
 	{
 		return $this->image_dimensions($path);
 	}
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function get_link($path)
+	{
+		return generate_board_url() . $this->root_path . $path;
```
Comment (gooof) [inline phpBB/phpbb/storage/adapter/local.php]: generate_board_url() does not need a root_path

Comment (rubencm) [reply phpBB/phpbb/storage/adapter/local.php]: Yes, sorry, it was just path, i forgot to change it

File: phpBB/phpbb/storage/adapter/adapter_interface.php [hunk 2]
```diff
@@ -85,4 +85,11 @@ public function rename($path_orig, $path_dest);
 	 * 													When the file cannot be copied
 	 */
 	public function copy($path_orig, $path_dest);
+
+	/**
+	 * Get direct link.
+	 *
+	 * @param string	$path	The file
+	 */
```
Comment (marc1706) [inline phpBB/phpbb/storage/adapter/adapter_interface.php]: Please also add return info.

File: phpBB/phpbb/storage/adapter/local.php [hunk 3]
```diff
@@ -331,4 +337,12 @@ public function file_image_height($path)
 	{
 		return $this->image_dimensions($path);
 	}
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function get_link($path)
+	{
+		return generate_board_url() . $this->path . $path;
```
Comment (marc1706) [inline phpBB/phpbb/storage/adapter/local.php]: Please use more distinctive names. Now it seems like your adding a path twice. The class' path looks to be the storage adapter's base path and the `$path` seems to be the file path relative to this folder but your variable naming is not clear enough for my taste.

File: phpBB/phpbb/storage/storage.php [hunk 4]
```diff
@@ -216,4 +216,14 @@ public function file_info($path)
 	{
 		return new file_info($this->adapter, $path);
 	}
+
+	/**
+	 * Get direct link
+	 *
+	 * @param string	$path	The file
+	 */
```
Comment (marc1706) [inline phpBB/phpbb/storage/storage.php]: Also add a `@return` here

File: phpBB/phpbb/storage/adapter/local.php (hunk 3 above)
Comment (rubencm) [reply phpBB/phpbb/storage/adapter/local.php]: Yes i know, is something i have to do in the future, but for me is the same name for different things

Comment (rubencm) [reply phpBB/phpbb/storage/adapter/local.php]: I change it a little on split uploaded files into subdirectories

**Comment by marc1706** [comment]
Please add tests for this newly added method.

**Comment by rubencm** [comment]
There is nothing to test

**Pull request merged**